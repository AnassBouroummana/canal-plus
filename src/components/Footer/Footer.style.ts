import styled from 'styled-components';
import { colorPalette, fontSize, getSpacing } from 'stylesheet';

export const StyledFooter = styled.div`
  margin-top: ${getSpacing(3)};
  text-align: center;
  border-top: 2px solid ${colorPalette.primary};
  color: ${colorPalette.white};
  padding: ${getSpacing(3)};
  font-size: ${fontSize.small};
`;
