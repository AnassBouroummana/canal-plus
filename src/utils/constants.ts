export const NODE_ENV = process.env.NODE_ENV || '';
export const IMG_BASE_PATH = 'https://image.tmdb.org/t/p/original/';
export const BASE_PATH = 'https://api.themoviedb.org/3/';
export const API_KEY = '92b418e837b833be308bbfb1fb2aca1e';
