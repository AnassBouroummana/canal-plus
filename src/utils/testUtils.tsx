import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Router } from 'react-router';
import { createBrowserHistory } from 'history';
import ConnectedIntlProvider from 'providers/ConnectedIntlProvider';
import theme from 'stylesheet';

export const AllTheProviders: React.FC<{}> = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <Router history={createBrowserHistory()}>
        <ConnectedIntlProvider>{children}</ConnectedIntlProvider>
      </Router>
    </ThemeProvider>
  );
};

const customRender = (
  ui: Parameters<typeof render>[0],
  options?: Parameters<typeof render>[1],
) => ({
  ...render(ui, { ...options, wrapper: AllTheProviders }),
  history: createBrowserHistory(),
});

const testSnapshot = (comp: React.ReactElement) => {
  const { container } = customRender(comp);
  expect(container).toMatchSnapshot();
};

export * from '@testing-library/react';
export { customRender as render, testSnapshot };
