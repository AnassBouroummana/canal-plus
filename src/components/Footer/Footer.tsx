import React from 'react';
import { FormattedMessage } from 'react-intl';
import { StyledFooter } from './Footer.style';

const Footer: React.FC = () => {
  return (
    <StyledFooter>
      <FormattedMessage id="footer.built-by" />
    </StyledFooter>
  );
};

export default Footer;
