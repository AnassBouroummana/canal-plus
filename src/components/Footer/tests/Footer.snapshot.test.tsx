import { render } from 'utils/testUtils';

import Footer from '../Footer';

/*
Snapshot tests allow you to easily lock the comportment of a component.
Given props, it renders the component and compares it to the saved snapshot.
*/
describe('[Snapshot] <Footer />', () => {
  it('should render a button with a label', () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
