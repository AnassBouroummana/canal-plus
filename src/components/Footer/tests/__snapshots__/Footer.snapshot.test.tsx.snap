// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Snapshot] <Footer /> should render a button with a label 1`] = `
<div>
  <div
    class="sc-bdvvtL hbLttv"
  >
    Build by Anass BOUROUMMANA // Alter Solutions
  </div>
</div>
`;
